### DISCUSSION TITLE
Ability to type without being obstructed by scrolling text above

### DISCUSSION THREAD

**Discussion by bradygaster** [spectreconsole/spectre.console#459] [Ideas]
Imagine a windows or web UI in which there is a scrolling listbox (that's a whole other feature I'd love to see, btw, but I digress...) at the top, and a text box at the bottom into which one can type. Like a chat app, or an old-school moo or mud. Something into which a command can be entered whilst "things are happening" above. I can type as fast as I need to, and the messages can come in as fast as they need to, but neither impacts the other's rendering. 

I'd love to have this feature, or some sort of ability to create scrolling experiences within borders or table cells. 

As abstractly-speaking as possible, I'd love to be able to interact with one segment of the screen whilst another (or multiple other) segments of the screen are responding to events happening in the app.  

**Describe the solution you'd like**
My MVP would be to have the interactive part of the console into which I can enter commands be unobstructed under the "display" region, where events in the application are causing messages to appear. Think "chat." 

**Describe alternatives you've considered**
Listboxes inside of table cells whilst having `Console.ReadLine`-like experiences in other cells simultaneously. 

**Additional context**
@reubenbond may have a screen shot from a sample he's building that demonstrates this idea.

**Comment by patriksvensson** [comment]
@bradygaster I think the interactivity you're asking for is easier to achieve with a TUI framework like Gui.cs (https://github.com/migueldeicaza/gui.cs). Still, we could definitely add a new widget that does what you ask for, that is, allowing input at the bottom of the console output while other renderable things are drawn on top.

@phil-scott-78 What do you think?

**Comment by bradygaster** [comment]
Yes one of the projects I'm building currently uses gui.cs. And I like it a lot, but there are features in Spectre I want to use MORE. :)

**Comment by bradygaster** [comment]
Congrats on the recent release. Was curious if you'd given this idea - or a workaround of some other type - consideration?
